config/environments/development.rb

```diff
@@ -54,3 +54,34 @@
   # Allow requests for all domains e.g. <app>.dev.gov.uk
   config.hosts.clear
 end
+
+# make discovery work over HTTP
+module OpenIDConnect
+  module Discovery
+    module Provider
+      class Config
+        class Resource
+          def initialize(uri)
+            @host = uri.host
+            @port = uri.port unless [80, 443].include?(uri.port)
+            @path = File.join uri.path, ".well-known/openid-configuration"
+            @scheme = uri.scheme
+            attr_missing!
+          end
+
+          def endpoint
```

Comment:
Not really :100:  clear what's going on here. What's the `SWD` where does it come from?

The commit message is nice and clear about what we're trying to achieve, but i'm not really sure how that's working here?

Reply:
I have to confess I just plagiarised this code from the [barking permit app](https://github.com/alphagov/govuk-accounts-demo-app/blob/a1585ef01bf804307b0c93bc0be2035946367c97/config/environments/development.rb#L67-L96).  I'd have to defer to @barrucadu for the thinking about how this works.

Reply:
I also plagiarised it

Reply:
Found [this comment](https://github.com/nov/openid_connect/issues/47#issuecomment-573701929) in my browser history.  Looks like that's where we got the code form, just with a bit of linting applied.

Reply:
:D cool, no prob looks like it's a fairly generic piece of code anyway so that's likely fitting.
I'll spin up a rails console and have a look at the classes for SWD to find out a bit more about it.

Thanks for the trail!